latex_access/polish_braille.py

```diff
@@ -0,0 +1,391 @@
+                translation = self.translate(translation,(rting[0]+arg[2],rting[1]+1))
+        return (translation,arg[1])
+
+
+    def sub(self,input,start,rting=()):
+        '''Translates subscripts to Polish braille.
+
+        Returns a touple, as above'''
+        arg=get_arg(input,start)
+        if arg[0].isdigit():
+            translation = u"ą"
+            for k in range(len(arg[0])) : translation+=self.table["L"+arg[0][k]]
+        elif not arg[0].isdigit() and len(arg[0]) ==1:
+            translation = u"ą" +self.translate(arg[0])
+        #With two lines below uncommented translation doesn't work as expected
+        #The conditional below always executes so numbers are never lowered
+        #if rting!=():
+            #for j in range(len(arg[0])): self.rt.append((rting[1]+j,rting[0]+arg[2]+j))
+        else:
+            if rting==():
+                translation = u"ą"+self.translate(arg[0]) + u"ą"
+            else:
+                translation = u"ą"+self.translate(arg[0],(rting[0]+arg[2],rting[1]+1)) + u"ą"
+        return (translation,arg[1])
+
+    def sqrt(self,input,start,rting=()):
+        '''Translatesroots in latex.
+
+        Returns a touple as above.'''
+        first_arg =latex_access.get_optional_arg(input, start)
+        if first_arg:
+            second_arg=get_arg(input, first_arg[1])
+            if first_arg[0] =="2":
+                translation =u"⠌⠆"
+            elif first_arg[0] =="3":
+                translation =u"⠌⠒"
+            elif first_arg[0].lower() =="n":
+                translation =u"⠌⠝"
+            elif first_arg[0].lower() =="x":
+                translation =u"⠌⠭"
+            elif first_arg[0] =="":
+                translation =u""
+            else:
+                translation =u"⠌"
+                for k in range(len(first_arg[0])) : translation+=self.table["L"+first_arg[0][k]]
+            if len(second_arg[0]) ==1 or second_arg[0].isdigit():
+                self.lastnumber=-1
+                translation +=u"ć" +self.translate(second_arg[0])
+            else:
+                self.lastnumber=-1
+                translation +=u"ć" +self.translate(second_arg[0])
+                translation +=u"ę"
+            return (translation,second_arg[1])
+        else:
+            arg=get_arg(input,start)
+            second_arg=""
+        if arg[0].isdigit() or len(arg[0])==1:
+            translation =u"ć" +arg[0] +u""
+        else:
+            translation =u"ć" +self.translate(arg[0]) +u"ę"
+        return (translation,arg[1])
+
+
+    def frac(self,input,start,rting=()):
+        '''Translates fractions into polish braille.
+
+        Returns touple as above'''
+        numerator=get_arg(input,start)
+        if numerator[1]<len(input):
+            denominator=get_arg(input,numerator[1])
+        else:
+            denominator=("",numerator[1],numerator[1])
+        if numerator[0].isdigit() and denominator[0].isdigit():
+            if input[start-6].isdigit() and start>5:
```

Comment:
These two conditions should be swapped i.e. we should first check if `start` is bigger than 5, and only if it is try to check if there is a digit before the fraction.